assemblyline_core/updater/helper.py

```diff
@@ -98,7 +98,7 @@ def get_latest_tag_for_service(service_config, system_config, logger):
     else:
         tag_name = f"{FRAMEWORK_VERSION}.{SYSTEM_VERSION}.0.{update_channel}0"
         for t in tags:
-            if update_channel in t:
+            if len(t.split(update_channel))==2 and len(t.split(".")) >= 2:
```

Comment:
Now I'm not too familiar with this functionality.... but in theory, if I had `t == "blah...stable"`, this would pass this `if`? Wouldn't a regex be more applicable because then we'd have more control over what input actually gets passed?

Reply:
I'm not too familiar with regex but would something like:
`\d[.]\d[.]\d[.]\w*\d`

work?

Reply:
Yes that would work, but I think we need to confirm what we're expecting to be valid before we design a REGEX. Is this variable representative of how we define a proper tag name `tag_name = f"{FRAMEWORK_VERSION}.{SYSTEM_VERSION}.0.{update_channel}0"` ?

Reply:
Yes, considering our tags have followed that schema.

This would probably be better because it would count the digit at the end: \d+[.]\d+[.]\d+[.]\D+\d+